src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/ApartmentFactory.java

```diff
@@ -0,0 +1,191 @@
+package io.github.oliviercailloux.y2018.apartments.apartment;
+
+import io.github.oliviercailloux.y2018.apartments.apartment.Apartment.Builder;
+import io.github.oliviercailloux.y2018.apartments.utils.JsonConvert;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Random;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A factory for creating Apartment objects.
+ * @author Clémence COUSIN & Gabriel GUISSET 
+ */
+public abstract class ApartmentFactory {
+
+	/** This constant allow us to know the maximum number of times we should try to 
+	 * reach the API address */
+	private static final int NB_MAX_RETRY = 5;
+
+	/** Logger whic allow us to know where the mistakes are. */
+	private static Logger LOGGER = LoggerFactory.getLogger(ApartmentFactory.class);
+
+	/** Random object which allow us to generate random numbers */
+	private static Random rand = new Random();
+
+	/** The URL of the API address. */
+	private static String urlApiAddress = "https://8n8iajahab.execute-api.us-east-1.amazonaws.com/default/RealRandomAdress";
```

Comment:
Addition of the final keyword and the convention for the constants
```suggestion
	private static final String URL_API_ADDRESS = "https://8n8iajahab.execute-api.us-east-1.amazonaws.com/default/RealRandomAdress";
```

Reply:
done in commit 790711bdb9648f9fafc7004c0e49661562b92b04

Reply:
i used auto formatting of eclipse and no changes were done.
Do you speak about indentation ? of only about attributes' pattern ?

Reply:
These were mainly spaces. It's just a question of responding to good practices.
For example :

```
if (1<= 3) // 1 space on the right and 0 on the left
Apartment.callFunction(param1,param2,param3) // need space between params
```

Concerning the formatting of the text with the indentation, I think that we will make a pull request specially for that in order to reformat the code of all the project

Reply:
done in commit 92d2a39ae2baf8437530dc00222a7cceaba2d7af